packages/lib/server/defaultResponder.ts

```diff
@@ -1,9 +1,17 @@
 import type { NextApiRequest, NextApiResponse } from "next";
 
+import { type TraceContext } from "@calcom/lib/tracing";
+import { TracedError } from "@calcom/lib/tracing/error";
+import { distributedTracing } from "@calcom/lib/tracing/factory";
+
 import { getServerErrorFromUnknown } from "./getServerErrorFromUnknown";
 import { performance } from "./perfObserver";
 
-type Handle<T> = (req: NextApiRequest, res: NextApiResponse) => Promise<T>;
+export interface TracedRequest extends NextApiRequest {
+  traceContext: TraceContext;
+}
+
+type Handle<T> = (req: TracedRequest, res: NextApiResponse) => Promise<T>;
```

Comment:
nit: We could introduced TracedResponse as well. For now, it would be just an alias of NextApiResponse